ignite/metrics/precision.py

```diff
@@ -30,43 +30,43 @@ def __init__(
                 )
 
         self._average = average
-        self._true_positives = None
-        self._positives = None
         self.eps = 1e-20
+        self._is_reduced = False
```

Comment:
Normally, `self._is_reduced = False` is called in `Metric.__init__`, so let's remove it from here.

Reply:
if I remove it, then I get a `mypy` issue and have to ignore it like I did before
```python
if not self._is_reduced:  # type: ignore[has-type] # mypy can't determine type
```

Reply:
Yeah, I do not understand why it looses the type. I checked with `reveal_type(self._is_reduced)` inside `_BaseClassification` and `Metric` and it is bool. But in `Precision` it becomes `Any`...

Reply:
what do you prefer? Define it in the `__init__` or add the `ignore`?

Reply:
OK, let's ignore it for this PR and fix if possible after.

Reply:
I think I found how to fix it:
```
            if not self._is_reduced:
                self._true_positives = idist.all_reduce(self._true_positives)  # type: ignore[arg-type, assignment]
                self._positives = idist.all_reduce(self._positives)  # type: ignore[arg-type, assignment]
                self._is_reduced = True  # type: bool

```

Reply:
you are right and it works, but I have no idea, why this fixes this problem?!